src/averageAges.js

```diff
@@ -52,7 +61,10 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  return people.filter(e => people.some(x => x.name === e.mother))
+    .filter(e => onlyWithSon !== undefined ? e.sex === 'm' : true)
+    .map(e => e.born - people.find(x => x.name === e.mother).born)
```

Comment:
.map(e => e.born - people.find(x => x.name === e.mother).born)

Nice usage of methods. I've been thinking for so long how to combine these two methods and still failed.